src/Utilities/Compiler/SymbolEqualityComparer.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis
+{
+#if !CODEANALYSIS_V3_OR_BETTER
```

Comment:
I had some exceptions saying cannot find `SymbolEqualityComparer` which made me realize that it's available only since v3. Adding a class matching the symbol to fallback to object equality seems the simplest solution as it will prevent pragmas everywhere in the code and make it simple when we drop support for older version to "bump" to the new version.

I am not entirely sure why but this symbol is not giving me what I would except so I end up with some warnings telling me that the symbol exists in 2 locations.

Any idea which compilation symbol to use to have the "proper" fix?

Reply:
> I am not entirely sure why but this symbol is not giving me what I would except so I end up with some warnings telling me that the symbol exists in 2 locations.


@Evangelink I'm not seeing anything about that in CI.

Reply:
I can no longer reproduce the issue. I am wondering if it was just some kind of non-refresh from VS.